build/Generator/ApiGenerator.php

```diff
@@ -217,6 +220,23 @@ private function generateInputClass(string $service, string $operationName, stri
             $this->inputClassRequestGetters($inputShape, $class, $operationName);
         }
 
+        // Add validate()
+        $namespace->addUse(InvalidArgument::class);
+        $requiredArray = '\'' . implode("', '", $requiredProperties) . '\'';
+
+        $validateBody = <<<PHP
```

Comment:
whats about?
```php
foreach ($requiredProperties as $property) {
   $validateBody.= "if (null === \$this->$property) throw ...;";
   if ($defintions[$property]['type'] === 'struct') {
     $validateBody.=' \$this->$property->validate();";
   }
}

Reply:
Oh, that is correct. I was looking to figure things out in the return type. 

I'll give it a try

Reply:
Fixed it!